apps/bfd-pipeline/bfd-pipeline-rif-load/src/main/java/gov/cms/bfd/pipeline/rif/load/RifLoader.java

```diff
@@ -596,6 +614,313 @@ private void processAsync(
+              beneficiaryRecord.getPartCPbpNumberOctId(),
+              beneficiaryRecord.getPartCPlanTypeOctCode(),
+              beneficiaryRecord.getPartDContractNumberOctId(),
+              beneficiaryRecord.getPartDLowIncomeCostShareGroupOctCode(),
+              beneficiaryRecord.getPartDPbpNumberOctId(),
+              beneficiaryRecord.getPartDRetireeDrugSubsidyOctInd(),
+              beneficiaryRecord.getPartDSegmentNumberOctId());
+
+      if (beneficiaryMonthly != null) {
+        currentYearBeneficiaryMonthly.add(beneficiaryMonthly);
+      }
+
+      beneficiaryMonthly =
+          getBeneficiaryMonthly(
+              beneficiaryRecord,
+              LocalDate.of(year, 11, 1),
+              beneficiaryRecord.getEntitlementBuyInNovInd(),
+              beneficiaryRecord.getFipsStateCntyNovCode(),
+              beneficiaryRecord.getHmoIndicatorNovInd(),
+              beneficiaryRecord.getMedicaidDualEligibilityNovCode(),
+              beneficiaryRecord.getMedicareStatusNovCode(),
+              beneficiaryRecord.getPartCContractNumberNovId(),
+              beneficiaryRecord.getPartCPbpNumberNovId(),
+              beneficiaryRecord.getPartCPlanTypeNovCode(),
+              beneficiaryRecord.getPartDContractNumberNovId(),
+              beneficiaryRecord.getPartDLowIncomeCostShareGroupNovCode(),
+              beneficiaryRecord.getPartDPbpNumberNovId(),
+              beneficiaryRecord.getPartDRetireeDrugSubsidyNovInd(),
+              beneficiaryRecord.getPartDSegmentNumberNovId());
+
+      if (beneficiaryMonthly != null) {
+        currentYearBeneficiaryMonthly.add(beneficiaryMonthly);
+      }
+
+      beneficiaryMonthly =
+          getBeneficiaryMonthly(
+              beneficiaryRecord,
+              LocalDate.of(year, 12, 1),
+              beneficiaryRecord.getEntitlementBuyInDecInd(),
+              beneficiaryRecord.getFipsStateCntyDecCode(),
+              beneficiaryRecord.getHmoIndicatorDecInd(),
+              beneficiaryRecord.getMedicaidDualEligibilityDecCode(),
+              beneficiaryRecord.getMedicareStatusDecCode(),
+              beneficiaryRecord.getPartCContractNumberDecId(),
+              beneficiaryRecord.getPartCPbpNumberDecId(),
+              beneficiaryRecord.getPartCPlanTypeDecCode(),
+              beneficiaryRecord.getPartDContractNumberDecId(),
+              beneficiaryRecord.getPartDLowIncomeCostShareGroupDecCode(),
+              beneficiaryRecord.getPartDPbpNumberDecId(),
+              beneficiaryRecord.getPartDRetireeDrugSubsidyDecInd(),
+              beneficiaryRecord.getPartDSegmentNumberDecId());
+
+      if (beneficiaryMonthly != null) {
+        currentYearBeneficiaryMonthly.add(beneficiaryMonthly);
+      }
+
+      if (currentYearBeneficiaryMonthly.size() > 0) {
+        List<BeneficiaryMonthly> currentBeneficiaryMonthlyWithUpdates;
+
+        Beneficiary beneficiaryFromDb =
+            entityManager.find(Beneficiary.class, beneficiaryRecord.getBeneficiaryId());
```

Comment:
I'm tired and can't quite think my way through it, but I think this might open us up to a race condition if we ever got a RIF file that listed the same bene multiople times? CSV records are processed in parallel, so there's a chance that you have two threads working on and trying to update the same bene record at the same time.

That'd be weird and bad, but perhaps worth thinking about and at least adding a "`TODO enforce RIF invariant elsewhere: no repeats of same record/PK in same RIF file allowed`".

Reply:
added to do